### PR TITLE
Enable code coverage, check styles, and spot bugs

### PR DESCRIPTION
**Pull request by lahirusamith**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline build.gradle:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: java.jdbc-native/build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline java.jdbc-native/build.gradle:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

File: java.jdbc-ballerina/build.gradle [hunk 3]
```diff
@@ -167,18 +177,33 @@ task ballerinaBuild {
...
     dependsOn(assemble)
     dependsOn(updateTomlFile)
+    dependsOn(initializeVariables)
     finalizedBy(revertTomlFile)
 
     inputs.dir file(project.projectDir)
+
+    def testParams = "--skip-tests"
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(':sql-ballerina:build')) {
```
Comment (BuddhiWathsala) [inline java.jdbc-ballerina/build.gradle:187]:
```suggestion
        if (graph.hasTask(':java.jdbc-ballerina:build')) {
```

File: java.jdbc-ballerina/build.gradle [hunk 4]
```diff
@@ -167,18 +177,33 @@ task ballerinaBuild {
...
+            ballerinaTest.enabled = false
+        }
+        if (graph.hasTask(':sql-ballerina:test')) {
```
Comment (BuddhiWathsala) [inline java.jdbc-ballerina/build.gradle:190]:
```suggestion
        if (graph.hasTask(':java.jdbc-ballerina:test')) {
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -41,10 +43,16 @@ jobs:
...
         with:
           name: Ballerina Internal Log Windows
           path: java.jdbc-ballerina/ballerina-internal.log
           if-no-files-found: ignore
+      - name: Archive Code Coverage JSON
+        uses: actions/upload-artifact@v2
+        with:
+          name: Code Coverage JSON
+          path: java.jdbc-ballerina/target/report/test_results.json
+          if-no-files-found: ignore
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request.yml:58]: We can remove this.

File: build.gradle [hunk 6]
```diff
@@ -16,10 +16,10 @@
...
-    id "com.github.johnrengelman.shadow" version "5.2.0"
-    id "de.undercouch.download" version "4.0.4"
-    id "net.researchgate.release" version "2.8.0"
+    id "com.github.spotbugs" version "${githubSpotbugsVersion}"
+    id "com.github.johnrengelman.shadow" version "${githubJohnrengelmanShadowVersion}"
+    id "de.undercouch.download" version "${underCouchDownloadVersion}"
+    id "net.researchgate.release" version "${researchgateReleaseVersion}"
 }
 
 ext.ballerinaLangVersion = project.ballerinaLangVersion
```
Comment (ThisaruGuruge) [inline build.gradle:25]:
```suggestion
```

File: java.jdbc-ballerina/build.gradle [hunk 7]
```diff
@@ -167,18 +177,33 @@ task ballerinaBuild {
...
     dependsOn(assemble)
     dependsOn(updateTomlFile)
+    dependsOn(initializeVariables)
     finalizedBy(revertTomlFile)
 
     inputs.dir file(project.projectDir)
+
+    def testParams = "--skip-tests"
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(':java.jdbc-ballerina:build')) {
```
Comment (ThisaruGuruge) [inline java.jdbc-ballerina/build.gradle:187]:
```suggestion
        if (graph.hasTask(':java.jdbc-ballerina:build') || graph.hasTask(':java.jdbc-ballerina:publish')) {
```

**Review by ThisaruGuruge — APPROVED** [review]

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**